tests/data_templates/test_field_definitions.py

```diff
@@ -0,0 +1,184 @@
+import os
+import json
+import pytest
+import pandas
+
+from marshmallow import ValidationError
+from pprint import pprint
+
+from creator.data_templates.models import TemplateVersion
+from creator.data_templates.field_definitions_schema import (
+    coerce_number,
+    coerce_bool,
+    FieldDefinitionSchema,
+    FieldDefinitionsSchema
+)
+
+
+@pytest.mark.parametrize(
+    "in_value, expected_out",
+    [
+        ("0.0", 0.0),
+        ("0", 0),
+        ("foo", "foo"),
+        ("1.234", 1.234),
+        ("200", 200),
+        (1.234, 1.234),
+        (200, 200),
+        (10.0, 10),
```

Comment:
just curious but why does `0.0` -> `0.0` but `10.0` -> `10`?

Reply:
Ha, good question. Not sure, maybe I have a bug in the code

Reply:
Ok, I fixed it. It was correctly coercing "0.0" -> 0.0 and "10.0" -> 10.0 but it would convert 0.0 -> 0 and 10.0 ->10 which is probably undesired. Now it will leave ints/floats as is and only try to coerce str values. So "0.0" -> 0.0 and "10.0" ->10.0